controller/recurring_job_controller.go

```diff
@@ -380,7 +381,16 @@ func (control *RecurringJobController) checkAndUpdateCronJob(cronJob, appliedCro
 
 func (control *RecurringJobController) newCronJob(recurringJob *longhorn.RecurringJob) (*batchv1beta1.CronJob, error) {
 	backoffLimit := int32(CronJobBackoffLimit)
-	successfulJobsHistoryLimit := int32(CronJobSuccessfulJobsHistoryLimit)
+	settingSuccessfulJobsHistoryLimit, err := control.ds.GetSettingAsInt(types.SettingNameRecurringSuccessfulJobsHistoryLimit)
+	if err != nil {
+		return nil, werror.NewInvalidError(fmt.Sprintf("failed to get setting of RecurringSuccessfulJobsHistoryLimit: %v", err), "")
```

Comment:
It's better not to use `webhook/error` in controllers. At least this is not an HTTP error...

Reply:
Ray, you can learn from other methods instead to use the consistent way like `errors.Wrapf`

Reply:
Okay, I use the error library to wrap the error. And I have fixed the bug.
https://github.com/longhorn/longhorn-manager/blob/a9c3853113e0d70b114260aeeaba020f3f2d9240/controller/recurring_job_controller.go#L383-L392